### PR TITLE
Adds the gemset separator tests 

### PR DESCRIPTION
**Pull request by dacat**
The tests currently fail because the current version of rvm doesn't support @@.

you see the build here https://travis-ci.org/github/dacat/rvm-test/builds/675076020

### CODE REVIEW

**Comment by eregon** [comment]
Hello, could you make the PR against https://github.com/rvm/rvm?
This repo's CI doesn't match the CI for rvm/rvm, and it's quite old, so it's not good to test changes.

As mentioned on Slack I think we should archive this repo and add a note in the README to clarify that.

**Comment by eregon** [comment]
I added a mention in the README in 5e3fc6f2c0c74d600ae071dbc9e70c031544499b

@mpapis @pkuczynski OK to archive this repository to make it clear PRs should be made to rvm/rvm instead?

**Comment by pkuczynski** [comment]
@eregon this repo is used by `rvm/rvm` using git reference. I discussed this with @mpapis a while ago about moving tests into rvm, but he complained about release complexity...

File: .travis.yml [hunk 1]
```diff
@@ -8,17 +8,18 @@ before_install:
...
+  - RVM_GEMSET_SEPARATOR='@@' TEST=other/gemset_separator_tests_preprocessed/*
+-notifications:
+-  irc:
+-    channels:
+-      - "irc.freenode.org#rvm-test"
+-  email:
+-    recipients:
+-      - mpapis@gmail.com
+-      - me@deryldoucette.com
+-    on_failure: change
```
Comment (mpapis) [inline .travis.yml:25]: `- ` prefix does not look right

**Review by mpapis — CHANGES_REQUESTED** [review]
I do not like the idea of pre-processing the tests, bash part does evaluation of environment variables, the missing part is evaluating the test code in comments.

Best approach would be to modify `tf` to allow evaluation of variables in tests. Then common tests could be modified like: 
```bash
rvm use 2.7.0${RVM_GEMSET_SEPARATOR} # match=/Using .*$RVM_GEMSET_SEPARATOR$/
```
where `${RVM_GEMSET_SEPARATOR}` is evaluated by shell and `$RVM_GEMSET_SEPARATOR$` is evaluated by `tf`.

This way the tests could be made configurable and we could run same test twice. 1st with `@` separator and 2nd with `@@` separator.

**Comment by eregon** [comment]
@pkuczynski Ah, it's a git subtree: https://github.com/rvm/rvm/issues/4663#issuecomment-524563043
(it used to be a git submodule but that changed in that issue)

I would still like to move this PR, the CI of this repository is completely outdated and merging here will not actual run the new tests when changing rvm/rvm.
The git subtree allows bidirectional synchronization, so I would suggest that all rvm-test changes are done in rvm, and this repository simply serves as history.

Given the history of these tests is probably not that valuable and the effort is synchronize is probably not worth it, I would like to still suggest to just archive this repo and let tests live in rvm/rvm, which is simpler for everybody as they are already there.

**Comment by eregon** [comment]
Let's move the review to https://github.com/rvm/rvm/pull/4832

Implementation and tests together please, otherwise it's a nightmare and we've anyway been changing rvm-test in `rvm/rvm` directly for a while.

**Pull request closed without merging**